src/App.js

```diff
@@ -22,87 +20,74 @@ class App extends Component {
     getAllMovies()
       .then((response) => {
         this.setState({ fetchStatus: response.status });
-
-        if (!response.ok) {
-          throw new Error();
-        }
-
-        return response.json();
+        return handleErrors(response);
       })
       .then((movieFetchData) => this.setState({ movies: movieFetchData.movies }))
-      .catch((err) => this.setState( {fetchError: true }));
+      .catch(() => this.setState({ fetchError: true }));
   }
 
   displayMovieDetail = (id) => {
     // display placeholder?
     getSelectedMovie(id)
       .then((response) => {
         this.setState({ fetchStatus: response.status });
-
-        if (!response.ok) {
-          throw new Error();
-        }
-
-        return response.json();
+        return handleErrors(response);
       })
       .then((movieData) => this.setState({ selectedMovieDetails: movieData.movie }))
-      .catch((err) => this.setState( {fetchError: true }));
+      .catch(() => this.setState({ fetchError: true }))
+      // .then(this.setState({ fetchError: true, fetchStatus: 404 })) //to test errors on click
   }
 
   resetHome = () => {
-    this.setState({ selectedMovieDetails: {} });
-  }
-
-  render() {
-    // return this.renderError(404); // To force render of error state
-
-    if (this.state.fetchError) {
-      return this.renderError(this.state.fetchStatus);
-    }
-
-    return (
-      <div>
-        <Nav resetHome={this.resetHome} />
-        {!this.state.selectedMovieDetails.id
-          && (
-            <main className="home-page">
-              <Carousel movies={this.state.movies} displayMovieDetail={this.displayMovieDetail} />
-              <Movies movies={this.state.movies} displayMovieDetail={this.displayMovieDetail} />
-            </main>
-          )}
-        {this.state.selectedMovieDetails.id
-          && (
-            <main className="detail-page">
-              <MovieDetail details={this.state.selectedMovieDetails} />
-            </main>
-          )}
-      </div>
-    );
+    this.setState({
+      selectedMovieDetails: {},
+      fetchError: false,
+      fetchStatus: 0
+    });
   }
 
-  renderError(fetchStatus) {
+  renderError = (fetchStatus) => {
     let message;
 
     switch (fetchStatus) {
-      case 0: // Fetch failed entirely (0 is initial status)
-        message = 'Oops! Something went wrong. Please check your internet connection.'
+      case 0:
+        message = 'Oops! Something went wrong. Please check your internet connection.';
         break;
       case 404:
-        message = 'Page not found.'
+        message = 'Page not found.';
         break;
       case 422:
-        message = 'Invalid request.'
+        message = 'Invalid request.';
         break;
       default:
         message = 'Oops! Something went wrong. Please try again.';
     }
 
+    return (
+      <Error message={message} />
+    );
+  }
+
+  toggleView = () => {
```

Comment:
I like this refactor a lot. It makes sense to have a `<Main />` component with conditional rendering.